components/NewReplySection/index.js

```diff
@@ -0,0 +1,160 @@
+          id
+          articleReplies {
+            ...RelatedArticleReplyData
+          }
+        }
+      }
+    }
+  }
+  ${RelatedReplies.fragments.RelatedArticleReplyData}
+`;
+
+const CREATE_REPLY = gql`
+  mutation CreateReplyInArticlePage(
+    $articleId: String!
+    $text: String!
+    $type: ReplyTypeEnum!
+    $reference: String
+  ) {
+    CreateReply(
+      articleId: $articleId
+      text: $text
+      type: $type
+      reference: $reference
+    ) {
+      id
+    }
+  }
+`;
+
+const CONNECT_REPLY = gql`
+  mutation ConnectReplyInArticlePage($articleId: String!, $replyId: String!) {
+    CreateArticleReply(articleId: $articleId, replyId: $replyId) {
+      articleId
+    }
+  }
+`;
+
+const NewReplySection = withReplyFormContext(
+  ({
+    article,
+    existingReplyIds,
+    relatedArticles,
+    onSubmissionComplete,
+    onClose,
+  }) => {
+    const { fields, handlers } = useContext(ReplyFormContext);
+    const [flashMessage, setFlashMessage] = useState(0);
+    const currentUser = useCurrentUser();
+
+    const [createReply, { loading: creatingReply }] = useMutation(
+      CREATE_REPLY,
+      {
+        refetchQueries: ['LoadArticlePage'],
+        awaitRefetchQueries: true,
+        onCompleted() {
+          onSubmissionComplete(); // Notify upper component of submission
+          handlers.clear();
+          onClose();
+          setFlashMessage(t`Your reply has been submitted.`);
+        },
+        onError(error) {
+          console.error(error);
+          setFlashMessage(error.toString());
+        },
+      }
+    );
+    const [connectReply, { loading: connectingReply }] = useMutation(
+      CONNECT_REPLY,
+      {
+        refetchQueries: ['LoadArticlePage'],
+        awaitRefetchQueries: true,
+        onCompleted() {
+          onSubmissionComplete(); // Notify upper component of submission
+          setFlashMessage(t`Your have attached the reply to this message.`);
+        },
+        onError(error) {
+          console.error(error);
+          setFlashMessage(error.toString());
+        },
+      }
+    );
+
+    const handleSubmit = useCallback(
+      e => {
+        const { replyType: type, reference, text } = fields;
+        e.preventDefault(); // prevent reload
+        if (creatingReply) return;
+        createReply({
+          variables: { type, reference, text, articleId: article.id },
+        });
+      },
+      [createReply]
+    );
+
+    const relatedArticleReplies = getDedupedArticleReplies(
+      relatedArticles,
+      existingReplyIds
+    );
+
+    const handleConnect = replyId => {
+      connectReply({ variables: { articleId: article.id, replyId } });
+    };
+
+    if (!currentUser) {
+      return <p>{t`Please login first.`}</p>;
+    }
+
+    const sharedProps = {
+      relatedArticleReplies,
+      handleSubmit,
+      handleConnect,
+      connectingReply,
+      creatingReply,
+      existingReplyIds,
+    };
+
+    return (
+      <form onSubmit={handleSubmit}>
```

Comment:
Nested `form` is introduced here, it's not valid DOM structure. Please remove one of the nesting form.

![image](https://user-images.githubusercontent.com/108608/82548180-80b10400-9b8d-11ea-8586-e688044708a9.png)